script.js

```diff
@@ -1,4 +1,207 @@
+  }
+
+  playerProfiles[currPlayer].diceRolls = newDiceRolls;
+
+  return newDiceRolls;
+};
+
+var concatenate2Numbers = function (num1, num2) {
+  return Number(String(num1) + String(num2));
+};
+
+var getPlayerNumberAutomatically = function () {
+  var diceArray = playerProfiles[currPlayer].diceRolls;
+  var playerNum;
+
+  diceArray.sort(function (a, b) {
+    return a - b;
+  });
+
+  while (diceArray.length > 0) {
+    var largestNum = diceArray.pop();
+    if (playerNum == null) {
+      playerNum = largestNum;
+    } else {
+      playerNum = concatenate2Numbers(playerNum, largestNum);
+    }
+  }
+  playerProfiles[currPlayer].diceNum = playerNum;
+  return playerNum;
+};
+
+var determineWinner = function () {
+  if (roundWinner == null) {
+    roundWinner = playerProfiles[currPlayer];
+  } else if (playerProfiles[currPlayer].diceNum > roundWinner.diceNum) {
+    roundWinner = playerProfiles[currPlayer];
+  }
+};
+
+var addNumToRunningScore = function (roundScore) {
+  playerProfiles[currPlayer].score += roundScore;
+};
+
+var createLeaderBoardOutput = function () {
+  var leaderBoardOutput = "Leaderboard: ";
+
+  playerProfiles.sort((a, b) => b.score - a.score);
+  for (var i = 0; i < playerProfiles.length; i += 1) {
+    leaderBoardOutput +=
+      "<br> Player " + playerProfiles[i].id + ": " + playerProfiles[i].score;
+  }
+  return leaderBoardOutput;
+};
+
+var resetGame = function () {
+  currPlayer = 0;
+  gameMode = GAME_MODE_CHOOSE_NUM_DICE;
+  roundWinner = null;
+};
+
+var createDiceRollInfoMsg = function (diceRollsArr) {
+  var outputMsg =
+    "Welcome Player " +
+    playerProfiles[currPlayer].id +
+    ".<br> Your dice rolls are:";
+
+  for (var i = 0; i < diceRollsArr.length; i += 1) {
+    outputMsg += "<br> Dice " + (i + 1) + ": " + diceRollsArr[i];
+  }
+  return outputMsg;
+};
+
+var createPlayerProfiles = function () {
+  for (var i = 0; i < numPlayers; i += 1) {
+    playerProfiles.push({ id: i + 1, diceRolls: [], diceNum: 0, score: 0 });
+  }
+};
+var totalReset = function () {
+  currPlayer = 0;
+  gameMode = GAME_MODE_CHOOSE_NUM_PLAYERS;
+  roundWinner = null;
+  playerProfiles = [];
+  numPlayers = 0;
+  numDiceChosen = 0;
+};
+//
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  console.log(gameMode);
+  console.log(currPlayer);
+  if (gameMode == GAME_MODE_CHOOSE_NUM_PLAYERS) {
+    if (isNaN(input) == true) {
+      return "Please enter a number between 2 players to 5 players";
+    }
+    if (Number(input) < 2) {
+      return "At least 2 players to play! ";
+    }
+    if (Number(input) > 5) {
+      return "Too crowded! Max 5 players, please. ";
+    } else numPlayers = Number(input);
+
+    createPlayerProfiles();
+
+    gameMode = GAME_MODE_CHOOSE_NUM_DICE;
+    return (
+      "There are " +
+      numPlayers +
+      " players in this game. Please enter how many dice you would like to play with."
+    );
+  }
+
+  if (gameMode == GAME_MODE_CHOOSE_NUM_DICE) {
```

Comment:
I noticed you're only using if statements in your main function. Try to use else if where appropriate as well. Having multiple ifs like this signals to the reader that more than 1 if could potentially happen. But in your case, that's not what is really happening